### DISCUSSION TITLE
Vite build with sourcemaps

### DISCUSSION THREAD

**Discussion by stylecramper** [codemagic-ci-cd/codemagic-docs#2765] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Ionic/Capacitor

### Steps to reproduce

Our codemagic.yaml includes a script that uses vite to build artifacts using the command:
`vite build --sourcemap true`
If I run this command locally, the JS files and their associated sourcemap files are built into the ./dist/assets directory.

### Expected results

I expect the Codemagic build to do the same thing I'm seeing locally: build the JS files with their associated sourcemap files.

### Actual results

When the codemagic script runs the same command, I can see in the build overview that only the JS files - and not the sourcemap files - are built.

### Build id (optional)

66ff15f0704a7c9066add214

**Comment by dam-ease** [comment]
Hello @stylecramper.
I did a quick google search of the issue and can see a similar/related github issue here: https://github.com/remix-run/remix/issues/8935 which has been fixed with an update, do you specify the version of `vite` being used in any file which is included in your `.gitignore`? Also, please can you try with the newer version as suggested in that thread?

**Reply by stylecramper** [reply]
We only specify the version of vite in package.json. It looks like the newer version they're recommending is a newer version of remix-run? - which we're not using.

**Reply by dam-ease** [reply]
Hey @stylecramper
Oh, I see, I read that as they had updated remix-run to use a newer version of vite.
I went through the vite [docs](https://v3.vitejs.dev/config/build-options.html#build-sourcemap) and see it's worth confirming if build.options is set true in your `vite.config.js` and also the options `server` and `client`, I'm not so experienced with using vite but is there a possibility of running the `vite build --sourcemap true` command as a separate script in your yaml file with `--verbose` so we can get full logs?

**Reply by stylecramper** [reply]
Thanks @dam-ease , we have set `build.sourcemap` to true in our vite config. I'll explore using verbose

**Reply by stylecramper** [reply]
@dam-ease it looks like the [default for setting log verbosity level](https://v3.vitejs.dev/config/shared-options.html#loglevel) is 'info', not sure we can get anything better than that. I can already see the log output for every single file built, and that's how I know it's not producing sourcemap files

**Reply by stylecramper** [reply]
@dam-ease after another look, I think I may be wrong. I downloaded the log from the build and I see lines like this:
```
dist/assets/QuotesPage-BLg_XL1g.js                                                             4.17 kB │ gzip:   2.02 kB │ map:     11.16 kB
dist/assets/QuestionnairesPage-D97STp4Q.js                                                     4.42 kB │ gzip:   2.05 kB │ map:     11.95 kB
```
In the build overview I couldn't see the `map` items (it's VERY dark grey against slightly darker grey, didn't even see it).
Would you say this means the sourcemap files WERE built alongside the JS files? The next step `Upload source maps to Sentry` still finishes with "Nothing to upload"

**Reply by dam-ease** [reply]
Hey @stylecramper,
Yes, from this logs, it seems the source map files are being built alongside the `.js` files. Is this your expected behaviour with your configuration or does it build as a separate .map file locally? If this is expected, hence the issue seems to lie with uploading the source maps to Sentry, I went through the Sentry docs and I found that to add a dist to your uploaded artifacts, you need to run the sourcemaps upload command with the additional --dist option. As shown in the docs here: https://docs.sentry.io/platforms/javascript/sourcemaps/uploading/cli/#associating-dist-with-artifact-bundle
Please can you try below suggestions to see if it helps in this case:
1. Install Sentry dependency using the script below
```
scripts:
    - name: Install Sentry dependency
      script: | 
        npm install --save @sentry/react-native
```
2. Specify artifacts path: In this case should be the path should either be the `.js` file or the `.map` file in the artifacts section.

**Reply by dam-ease** [reply]
Hey @stylecramper,
Any update for me on this please?
If there is anything else I can assist you with, please don't hesitate to let me know.

**Comment by stylecramper** [comment]
Hi Damilola,

Another dev looked into it and determined that our script was missing the info from the .env file, it's working now. Thanks for the time and attention.

-Matt
________________________________

**Reply by dam-ease** [reply]
Glad it's all good now @stylecramper!
